src/core/crypto.js

```diff
@@ -995,189 +954,171 @@ var AES128Cipher = (function AES128CipherClosure() {
         result[result.length - 1] = lastBlock.subarray(0, 16 - psLen);
       }
     }
-    var output = new Uint8Array(outputLength);
+    let output = new Uint8Array(outputLength);
     for (i = 0, j = 0, ii = result.length; i < ii; ++i, j += 16) {
       output.set(result[i], j);
     }
     return output;
   }
 
-  AES128Cipher.prototype = {
-    decryptBlock(data, finalize) {
-      var i, sourceLength = data.length;
-      var buffer = this.buffer, bufferLength = this.bufferPosition;
-      // waiting for IV values -- they are at the start of the stream
+  decryptBlock(data, finalize, iv = null) {
+    let i, sourceLength = data.length;
+    let buffer = this.buffer, bufferLength = this.bufferPosition;
+    // if not supplied an IV wait for IV values
+    // they are at the start of the stream
+    if (iv) {
+      this.iv = iv;
+    } else {
       for (i = 0; bufferLength < 16 && i < sourceLength; ++i, ++bufferLength) {
         buffer[bufferLength] = data[i];
       }
-      if (bufferLength < 16) {
-        // need more data
+      if (bufferLength < 16) { // Need more data.
         this.bufferLength = bufferLength;
         return new Uint8Array([]);
```

Comment:
Pre-existing nit: This looks weird, let's use `new Uint8Array(0);` here and elsewhere instead.